### ISSUE TITLE
Use parallelism for Population

### ISSUE THREAD

**Issue by turion** [tweag/monad-bayes#247]
It seems to me that all inference using `Population` is done single-threaded. But it is a perfect target for trivial parallelization, because all the calculations are independent in a list. One could easily use e.g. https://hackage.haskell.org/package/parallel to make use of several cores.

**Comment by reubenharry** [comment]
Agreed. The caveat (as discussed) being that resampling is decidedly not parallel, so the advantage depends on the rate of the resampling being reasonably low. That said, I think parallelization here would likely be an easy performance win.
